### PR TITLE
nlr - mips port

### PR DESCRIPTION
**Pull request by janwilleke**
Add nlr support for mips architecture

### CODE REVIEW

File: py/nlrmips.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/mpstate.h"
+
+#if MICROPY_NLR_MIPS
+
+#undef nlr_push
```
Comment (dpgeorge) [inline py/nlrmips.c:31]: Is this undef needed?

File: py/nlrmips.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#include <stdio.h>
```
Comment (dpgeorge) [inline py/nlrmips.c:32]: Is this include needed?

File: py/py.cmake [hunk 3]
```diff
@@ -54,6 +54,7 @@ set(MICROPY_SOURCE_PY
     ${MICROPY_PY_DIR}/nativeglue.c
     ${MICROPY_PY_DIR}/nlr.c
     ${MICROPY_PY_DIR}/nlrpowerpc.c
+    ${MICROPY_PY_DIR}/nlrmips.c
```
Comment (dpgeorge) [inline py/py.cmake:57]: Please sort in alphabetical order, ie before nlrpowerpc.c

File: py/py.mk [hunk 4]
```diff
@@ -59,6 +59,7 @@ PY_CORE_O_BASENAME = $(addprefix py/,\
 	nlrthumb.o \
 	nlraarch64.o \
 	nlrpowerpc.o \
+	nlrmips.o \
```
Comment (dpgeorge) [inline py/py.mk:62]: please sort

**Review by dpgeorge — CHANGES_REQUESTED** [review]
Thanks for the contribution!

How did you test this?

**Comment by janwilleke** [comment]
> 

Thank you your your review, I improved are remarks ;-)

As I include micropython into my pic32mz projekt, the linker was missing setjmp, thus I had to fix this, and as maybe other people are using mips too.... 
So its been tested with qemu (as described in the patch) and on pic32mz real hardware.

**Comment by dpgeorge** [comment]
> So its been tested with qemu (as described in the patch) and on pic32mz real hardware.

Ok great!  I didn't read the commit message, but now I have and it does explain it well.

And I forgot that CI does run a MIPS build and tests, using the unix port.  So I wonder if that build now picks up this nlr implementation rather than setjmp.  Need to check that somehow.

**Comment by dpgeorge** [comment]
> And I forgot that CI does run a MIPS build and tests, using the unix port. So I wonder if that build now picks up this nlr implementation rather than setjmp. Need to check that somehow.

I checked this and the CI does use this new nlr code.

**Comment by dpgeorge** [comment]
Rebased and merged in 043dc4dd0cc4679fca96a2d36f1bf7cf1656606f and 40a3aa709c874f1eb3cabd6d8a393ea8042d976c

**Pull request closed without merging**